src/openqaoa-qiskit/backends/qaoa_qiskit_sim.py

```diff
@@ -152,10 +156,15 @@ def parametric_qaoa_circuit(self) -> QuantumCircuit:
 
         self.qiskit_parameter_list = []
         for each_gate in self.abstract_circuit:
-            angle_param = Parameter(str(each_gate.pauli_label))
-            self.qiskit_parameter_list.append(angle_param)
-            each_gate.rotation_angle = angle_param
-            if type(each_gate) in self.QISKIT_GATEMAP_LIBRARY:
+            # if gate is of type mixer or cost gate, assign parameter to it
+            if each_gate.gate_label.type.value in ["MIXER", "COST"]:
+                angle_param = Parameter(each_gate.gate_label.__repr__())
```

Comment:
I think it is cleaner if instead of `each_gate.gate_label.__repr__()` we use `str(each_gate.gate_label)`

Reply:
Maybe slightly more accurate to do `repr(each_gate.gate_label)` instead since `str(each_gate.gate_label)` tries to return the implementation of `__str__` before `__repr__`. Though `__str__` is not implementated for the GateLabel class, it might be in the future~~

Reply:
agreed with Shahidee. `str(abc)` would return abc.__str__() as far as I know

Reply:
since the two lines are functionally equivalent, we can change this in a later iteration. Resolving this for now